sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1126 @@
+     *
+     * @param inputs A list of {@link TextDocumentInput inputs/documents} to be analyzed.
+     * @return A {@link Mono} containing the {@link DocumentResultCollection batch} of the
+     * {@link NamedEntityResult named entity}.
+     * @throws NullPointerException if {@code inputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DocumentResultCollection<NamedEntityResult>> recognizeBatchEntities(List<TextDocumentInput> inputs) {
+        return recognizeBatchEntitiesWithResponse(inputs, null).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * TODO (shawn): add doc
+     *
+     * @param inputs A list of {@link TextDocumentInput inputs/documents}  to be analyzed.
+     * @param options TODO (shawn): add doc
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * {@link DocumentResultCollection batch} of {@link NamedEntityResult named entity}.
+     * @throws NullPointerException if {@code inputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DocumentResultCollection<NamedEntityResult>>> recognizeBatchEntitiesWithResponse(
+        List<TextDocumentInput> inputs, TextAnalyticsRequestOptions options) {
+        try {
+            return withContext(context -> recognizeBatchEntitiesWithResponse(inputs, options, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DocumentResultCollection<NamedEntityResult>>> recognizeBatchEntitiesWithResponse(
+        List<TextDocumentInput> document, TextAnalyticsRequestOptions options, Context context) {
+        final MultiLanguageBatchInput batchInput = new MultiLanguageBatchInput().setDocuments(document);
+        return service.entitiesRecognitionGeneralWithRestResponseAsync(
+            batchInput,
+            options == null ? null : options.getModelVersion(),
+            options == null ? null : options.showStatistics(), context)
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of named entities input - {}", batchInput))
+            .doOnSuccess(response -> logger.info("A batch of named entities output - {}", batchInput))
+            .doOnError(error -> logger.warning("Failed to named entities - {}", batchInput))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    private DocumentResultCollection<NamedEntityResult> toDocumentResultCollection(
+        final EntitiesResult entitiesResult) {
+        return new DocumentResultCollection<>(getDocumentNamedEntities(entitiesResult),
+            entitiesResult.getModelVersion(), entitiesResult.getStatistics());
+    }
+
+    private List<NamedEntityResult> getDocumentNamedEntities(final EntitiesResult entitiesResult) {
+        Stream<NamedEntityResult> validDocumentList = entitiesResult.getDocuments().stream()
```

Comment:
Same here. can we use a foreach loop rather than two functions that only get called once.